### ISSUE TITLE
[feature-request] Parameter transformation

### ISSUE THREAD

**Issue by chenyu-psy** [GidonFrischkorn/bmm-implement-m3#43]
When I try to run the following model, it doesn't work with an error message and warning message:
1. Error message: argument is of length zero
2. Warning message: You have not provided default_priors for at least one parameter in the model. Default priors will be specified internally based on the provided link function. Please check if the used priors are reasonable for your application 
```r
correct ~ a + cc + ct + b
other ~ a + b
colorDist ~ ra * a + rc * cc + b
tempDist ~ ra * a + rc * ct + b
otherDist ~ ra * a + b
NPL ~ b
ct ~ et * cc
a ~ 0 + condition + (0 + condition || participant)
cc ~ 0 + condition + (0 + condition || participant)
et ~ 0 + condition + (0 + condition || participant)
ra ~ 0 + condition + (0 + condition || participant)
rc ~ 0 + condition + (0 + condition || participant)
``` 

I notice that the issue is caused by the formula: `ct ~ et * cc`. Due to the current logic in BMM, `ct` is recognized as a parameter and should thereby be predicted by the data, and be provided priors. However, `ct` is more like a placeholder in the equation rather than a real parameter.

## Potential solution

1. Check whether both left side and right side of a formula are parameters.
2. If so, remove the parameter on the left side from the parameter list.

**Comment by chenyu-psy** [comment]
There could be another potential issue (I have not tested it yet): when I use some non-parameter values from the data set, will the model ask me to provide default priors for them? For instance, there is a column `t` in the dataset that represents the presentation time.

**Comment by chenyu-psy** [comment]
Here is a potential solution to the issue:

``` r

DVs <- c()
IVs <- c()
for (i in 1:length(formula)) {
  DVs <- c(DVs, all.vars(formula[[i]])[1])
  IVs <- c(IVs, all.vars(formula[[i]])[-1])
}
DVs <- unique(DVs)
IVs <- unique(IVs)

# response names
respNames <- setdiff(DVs, IVs)
DVparsNames <- intersect(DVs, IVs)
parNames <- setdiff(IVs, DVpars)

```

**Comment by GidonFrischkorn** [comment]
I have implemented some changes in this commit 12239b25e3a86f5167208dcb732e15fe298ef6ae and now the above mentioned formula did result in no errors when running the `check_data` and `check_formula` methods.

If you find time, it would be great to see if you can now estimate the model with these changes or if there are still problems when running the model.
